pkg/apperrors/errors.go

```diff
@@ -279,15 +292,15 @@ type InvalidEnumError struct {
 }
 
 // NewInvalidEnumError creates an invalid state error and wraps it as apprpriate
-func NewInvalidEnumError[EnumType ~string](err error, value EnumType, enumType string) InvalidEnumError {
-	return InvalidEnumError{
+func NewInvalidEnumError[EnumType ~string](err error, value EnumType, enumType string) *InvalidEnumError {
+	return &InvalidEnumError{
 		Err:   err,
 		Value: string(value),
 		Type:  enumType,
 	}
 }
 
-func (e InvalidEnumError) Error() string {
+func (e *InvalidEnumError) Error() string {
 	return fmt.Sprint("Value ", e.Value, " does not correspond to a valid enumeration for type ", e.Type)
 }
 
```

Comment:
Perhaps Unwrap should also take a pointer if the other receivers also are on pointers?

Reply:
Good catch, I'll update that.

Reply:
Implemented in https://github.com/CMSgov/easi-app/pull/2125/commits/8bf763aae7ff0625ee47bb10b3cc1625918e4739.